storage/diff_scheme/diff_manager.cpp

```diff
@@ -107,17 +107,24 @@ Status Manager::GetStatus() const
   return m_status;
 }
 
-void Manager::SetStatus(Status status)
+bool Manager::SizeFor(storage::TCountryId const & countryId, uint64_t & size) const
 {
   std::lock_guard<std::mutex> lock(m_mutex);
-  m_status = status;
+  if (!HasDiffForUnsafe(countryId))
+    return false;
+
+  size = m_diffs.at(countryId).m_size;
+  return true;
 }
 
-FileInfo const & Manager::InfoFor(storage::TCountryId const & countryId) const
+bool Manager::VersionFor(storage::TCountryId const & countryId, uint64_t & version) const
 {
   std::lock_guard<std::mutex> lock(m_mutex);
-  ASSERT(HasDiffForUnsafe(countryId), ());
-  return m_diffs.at(countryId);
+  if (!HasDiffForUnsafe(countryId))
```

Comment:
```c++
template <typename Fn>
bool WithDiff(string const & countryId, Fn && fn)
{
  std::lock_guard<std::mutex> lock(m_mutex);
  auto it = m_diffs.find(countryId);
  if (it == m_diffs.end())
    return false;
  fn(it->second);
  return true;
}
```

```
bool Manager::SizeFor(...)

{
  return WithDiff([&](Diff const & diff) { size = diff.m_size; });
}

bool Manager::VersionFor(...)
{
  return WithDiff(countryId, [&](Diff & diff) { version = diff.m_version; });
}
```

Reply:
Do you think it's a good idea not to check the status of diff manager?  Because HasDiffForUnsafe does this check as well. Also, we still need HasDiffFor as a public method, which uses HasDiffForUnsafe. So it seems it will be more redundant lines with that approach.